ui_framework/src/components/code_editor/code_editor.js

```diff
@@ -45,55 +54,97 @@ export class KuiCodeEditor extends Component {
   };
 
   startEditing = () => {
-    this.setState({ isHintActive: false });
+    this.setState({
+      isHintActive: false,
+    });
     this.aceEditor.editor.textInput.focus();
   }
 
   stopEditing() {
-    this.setState({ isHintActive: true });
+    this.setState({
+      isHintActive: true,
+      isEditing: false,
+    });
   }
 
   render() {
     const {
       width,
       height,
       onBlur, // eslint-disable-line no-unused-vars
+      isReadOnly,
+      setOptions,
+      cursorStart,
       ...rest,
     } = this.props;
 
-    const classes = classNames('kuiCodeEditorKeyboardHint', {
+    const classes = classNames('kuiCodeEditorWrapper', {
+      'kuiCodeEditorWrapper-isEditing': this.state.isEditing,
+    });
+
+    const promptClasses = classNames('kuiCodeEditorKeyboardHint', {
       'kuiCodeEditorKeyboardHint-isInactive': !this.state.isHintActive
     });
 
+    let filteredCursorStart;
+
+    const options = { ...setOptions };
+
+    if (isReadOnly) {
+      // Put the cursor at the beginning of the editor, so that it doesn't look like
+      // a prompt to begin typing.
+      filteredCursorStart = -1;
```

Comment:
Is this logic still needed now that css is used to hide the cursor?

Reply:
I think it still makes sense to put the cursor at the beginning for people who are using the keyboard to interact with the content. I'm guessing they'd want to copy/paste it and it makes sense to me intuitively to do that from start to end rather than the reverse.